### PR TITLE
Enable build_python on windows

### PR DESCRIPTION
**Pull request by mingzhe09088**
The PR aims to resolve issues related to BUILD_PYTHON and BUILD_TEST after FULL_CAFFE2 is removed on Windows.

### CODE REVIEW

**Comment by peterjc123** [comment]
These are the symbols that are not well annotated extracted from the build log.
```c++
error LNK2019: unresolved external symbol "__declspec(dllimport) public: char const * __cdecl caffe2::Registry<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::unique_ptr<class caffe2::OperatorBase,struct std::default_delete<class caffe2::OperatorBase> >,class caffe2::OperatorDef const &,class caffe2::Workspace *>::HelpMessage(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)const "  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl caffe2::Registry<class caffe2::TypeIdentifier,class std::unique_ptr<class caffe2::python::BlobFetcherBase,struct std::default_delete<class caffe2::python::BlobFetcherBase> > >::Registry<class caffe2::TypeIdentifier,class std::unique_ptr<class caffe2::python::BlobFetcherBase,struct std::default_delete<class caffe2::python::BlobFetcherBase> > >(void)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: class std::unique_ptr<class caffe2::python::BlobFetcherBase,struct std::default_delete<class caffe2::python::BlobFetcherBase> > __cdecl caffe2::Registry<class caffe2::TypeIdentifier,class std::unique_ptr<class caffe2::python::BlobFetcherBase,struct std::default_delete<class caffe2::python::BlobFetcherBase> > >::Create(class caffe2::TypeIdentifier const &)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl caffe2::Registry<enum at::DeviceType,class std::unique_ptr<class caffe2::python::BlobFeederBase,struct std::default_delete<class caffe2::python::BlobFeederBase> > >::Registry<enum at::DeviceType,class std::unique_ptr<class caffe2::python::BlobFeederBase,struct std::default_delete<class caffe2::python::BlobFeederBase> > >(void)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: class std::unique_ptr<class caffe2::python::BlobFeederBase,struct std::default_delete<class caffe2::python::BlobFeederBase> > __cdecl caffe2::Registry<enum at::DeviceType,class std::unique_ptr<class caffe2::python::BlobFeederBase,struct std::default_delete<class caffe2::python::BlobFeederBase> > >::Create(enum at::DeviceType const &)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: class std::vector<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::allocator<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > > > __cdecl caffe2::Registry<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::unique_ptr<class caffe2::db::DB,struct std::default_delete<class caffe2::db::DB> >,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,enum caffe2::db::Mode>::Keys(void)"  
error LNK2019: unresolved external symbol "public: __cdecl onnx_torch::ValueInfoProto::ValueInfoProto(void)"  
error LNK2019: unresolved external symbol "public: virtual __cdecl onnx_torch::ValueInfoProto::~ValueInfoProto(void)"  
error LNK2019: unresolved external symbol "private: void __cdecl onnx_torch::ValueInfoProto::InternalSwap(class onnx_torch::ValueInfoProto *)"  
error LNK2019: unresolved external symbol "public: __cdecl onnx_torch::TensorProto::TensorProto(void)"  
error LNK2019: unresolved external symbol "public: virtual __cdecl onnx_torch::TensorProto::~TensorProto(void)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl caffe2::Registerer<class caffe2::TypeIdentifier,class std::unique_ptr<class caffe2::python::BlobFetcherBase,struct std::default_delete<class caffe2::python::BlobFetcherBase> > >::Registerer<class caffe2::TypeIdentifier,class std::unique_ptr<class caffe2::python::BlobFetcherBase,struct std::default_delete<class caffe2::python::BlobFetcherBase> > >(class caffe2::TypeIdentifier const &,class caffe2::Registry<class caffe2::TypeIdentifier,class std::unique_ptr<class caffe2::python::BlobFetcherBase,struct std::default_delete<class caffe2::python::BlobFetcherBase> > > *,class std::function<class std::unique_ptr<class caffe2::python::BlobFetcherBase,struct std::default_delete<class caffe2::python::BlobFetcherBase> > __cdecl(void)>,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl caffe2::Registerer<enum at::DeviceType,class std::unique_ptr<class caffe2::python::BlobFeederBase,struct std::default_delete<class caffe2::python::BlobFeederBase> > >::Registerer<enum at::DeviceType,class std::unique_ptr<class caffe2::python::BlobFeederBase,struct std::default_delete<class caffe2::python::BlobFeederBase> > >(enum at::DeviceType const &,class caffe2::Registry<enum at::DeviceType,class std::unique_ptr<class caffe2::python::BlobFeederBase,struct std::default_delete<class caffe2::python::BlobFeederBase> > > *,class std::function<class std::unique_ptr<class caffe2::python::BlobFeederBase,struct std::default_delete<class caffe2::python::BlobFeederBase> > __cdecl(void)>,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: bool __cdecl caffe2::Registry<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::unique_ptr<class caffe2::GradientMakerBase,struct std::default_delete<class caffe2::GradientMakerBase> >,class caffe2::OperatorDef const &,class std::vector<struct caffe2::GradientWrapper,class std::allocator<struct caffe2::GradientWrapper> > const &>::Has(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: char const * __cdecl caffe2::Registry<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::unique_ptr<class caffe2::GradientMakerBase,struct std::default_delete<class caffe2::GradientMakerBase> >,class caffe2::OperatorDef const &,class std::vector<struct caffe2::GradientWrapper,class std::allocator<struct caffe2::GradientWrapper> > const &>::HelpMessage(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)const "  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: bool __cdecl caffe2::Registry<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::unique_ptr<class caffe2::OptimizationPass,struct std::default_delete<class caffe2::OptimizationPass> >,struct nom::repr::NNModule *>::Has(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: class std::unique_ptr<class caffe2::OptimizationPass,struct std::default_delete<class caffe2::OptimizationPass> > __cdecl caffe2::Registry<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::unique_ptr<class caffe2::OptimizationPass,struct std::default_delete<class caffe2::OptimizationPass> >,struct nom::repr::NNModule *>::Create(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,struct nom::repr::NNModule *)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: bool __cdecl caffe2::Registry<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::unique_ptr<class caffe2::WorkspaceOptimizationPass,struct std::default_delete<class caffe2::WorkspaceOptimizationPass> >,struct nom::repr::NNModule *,class caffe2::Workspace *>::Has(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: class std::unique_ptr<class caffe2::WorkspaceOptimizationPass,struct std::default_delete<class caffe2::WorkspaceOptimizationPass> > __cdecl caffe2::Registry<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::unique_ptr<class caffe2::WorkspaceOptimizationPass,struct std::default_delete<class caffe2::WorkspaceOptimizationPass> >,struct nom::repr::NNModule *,class caffe2::Workspace *>::Create(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,struct nom::repr::NNModule *,class caffe2::Workspace *)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: class std::unique_ptr<struct caffe2::python::PybindAddition,struct std::default_delete<struct caffe2::python::PybindAddition> > __cdecl caffe2::Registry<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::unique_ptr<struct caffe2::python::PybindAddition,struct std::default_delete<struct caffe2::python::PybindAddition> >,class pybind11::module &>::Create(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class pybind11::module &)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: class std::vector<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::allocator<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > > > __cdecl caffe2::Registry<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::unique_ptr<struct caffe2::python::PybindAddition,struct std::default_delete<struct caffe2::python::PybindAddition> >,class pybind11::module &>::Keys(void)"  
error LNK2019: unresolved external symbol "const onnx_torch::ValueInfoProto::`vftable'"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl nom::Graph<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::Graph<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >(void)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl nom::Graph<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::~Graph<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >(void)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: class std::vector<class nom::Node<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> > *,class std::allocator<class nom::Node<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> > *> > const __cdecl nom::Graph<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::getMutableNodes(void)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: class std::vector<class nom::Edge<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> > *,class std::allocator<class nom::Edge<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> > *> > const __cdecl nom::Graph<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::getMutableEdges(void)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl nom::Graph<class pybind11::object>::Graph<class pybind11::object>(void)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl nom::Graph<class pybind11::object>::~Graph<class pybind11::object>(void)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: class nom::Node<class pybind11::object> * __cdecl nom::Graph<class pybind11::object>::createNode(class pybind11::object &&)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: class nom::Edge<class pybind11::object> * __cdecl nom::Graph<class pybind11::object>::createEdge(class nom::Node<class pybind11::object> *,class nom::Node<class pybind11::object> *)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: class std::vector<class nom::Node<class pybind11::object> *,class std::allocator<class nom::Node<class pybind11::object> *> > const __cdecl nom::Graph<class pybind11::object>::getMutableNodes(void)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl nom::Node<class pybind11::object>::Node<class pybind11::object>(class nom::Node<class pybind11::object> &&)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: virtual __cdecl nom::Node<class pybind11::object>::~Node<class pybind11::object>(void)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: struct nom::repr::nn::NNNodeMatchCriteria __cdecl nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria>::getCriteria(void)const "  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl nom::Subgraph<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::Subgraph<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >(void)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: void __cdecl nom::Subgraph<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::addNode(class nom::Node<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> > *)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl nom::Subgraph<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::~Subgraph<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >(void)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl nom::Subgraph<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::Subgraph<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >(class nom::Subgraph<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> > const &)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl nom::Subgraph<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::Subgraph<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >(class nom::Subgraph<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> > &&)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl caffe2::Registerer<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::unique_ptr<struct caffe2::python::PybindAddition,struct std::default_delete<struct caffe2::python::PybindAddition> >,class pybind11::module &>::Registerer<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::unique_ptr<struct caffe2::python::PybindAddition,struct std::default_delete<struct caffe2::python::PybindAddition> >,class pybind11::module &>(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class caffe2::Registry<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::unique_ptr<struct caffe2::python::PybindAddition,struct std::default_delete<struct caffe2::python::PybindAddition> >,class pybind11::module &> *,class std::function<class std::unique_ptr<struct caffe2::python::PybindAddition,struct std::default_delete<struct caffe2::python::PybindAddition> > __cdecl(class pybind11::module &)>,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl nom::Graph<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper>::Graph<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper>(void)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl nom::Graph<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper>::~Graph<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper>(void)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: class nom::Node<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper> * __cdecl nom::Graph<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper>::createNode(struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper &&)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: class nom::Edge<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper> * __cdecl nom::Graph<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper>::createEdge(class nom::Node<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper> *,class nom::Node<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper> *,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: class std::vector<class nom::Node<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper> *,class std::allocator<class nom::Node<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper> *> > const __cdecl nom::Graph<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper>::getMutableNodes(void)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl nom::Subgraph<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper>::Subgraph<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper>(void)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: void __cdecl nom::Subgraph<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper>::addNode(class nom::Node<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper> *)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: void __cdecl nom::Subgraph<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper>::addEdge(class nom::Edge<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper> *)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl nom::Subgraph<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper>::~Subgraph<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper>(void)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl nom::Subgraph<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper>::Subgraph<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper>(class nom::Subgraph<struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::NodeWrapper,struct nom::algorithm::GraphWrapper<class nom::matcher::MatchNode<struct nom::repr::nn::NNNodeMatchCriteria> >::EdgeWrapper> const &)"  
error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl caffe2::Registry<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::unique_ptr<struct caffe2::python::PybindAddition,struct std::default_delete<struct caffe2::python::PybindAddition> >,class pybind11::module &>::Registry<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::unique_ptr<struct caffe2::python::PybindAddition,struct std::default_delete<struct caffe2::python::PybindAddition> >,class pybind11::module &>(void)"  
```

**Comment by orionr** [comment]
Waiting for https://github.com/pytorch/pytorch/pull/11622 to land and then I'll merge again

**Comment by orionr** [comment]
Actually - I take that back. Updating the submodule here.

**Comment by houseroad** [comment]
https://github.com/pytorch/pytorch/pull/11622 is landed :-)

**Comment by peterjc123** [comment]
I have skimmed the log. There seems to be only some remaining, like `const onnx_torch::ValueInfoProto::`vftable`. I guess that we'll need to restructure some of the code, moving the implementation part to the source file. (something like the solution 4 in this [post](https://www.codeproject.com/Questions/261491/Exporting-classes-from-a-DLL)) As for the last linking failure, I think it is probably caused by previous linking errors.

File: caffe2/core/qtensor.h [hunk 1]
```diff
@@ -14,7 +14,7 @@
 namespace caffe2 {
 
 template <class Context>
-class CAFFE2_API QTensor {
```
Comment (peterjc123) [inline caffe2/core/qtensor.h]: This one needs to be restructured, too. Otherwise the gcc tests won't pass.

**Pull request closed without merging**